awx/main/models/execution_environments.py

```diff
@@ -42,9 +42,11 @@ class Meta:
         on_delete=models.SET_NULL,
     )
     pull = models.CharField(
-        max_length=1024,
+        max_length=16,
         choices=PULL_CHOICES,
-        default='missing',
+        blank=True,
+        null=True,
```

Comment:
We should also ditch `null=True`.

Reply:
I'm going to remove that and interactive rebase, we don't need this many commits.